tutorials/plot_dics.py

```diff
@@ -0,0 +1,276 @@
+ax = axes[1][0]
+ax.plot(f[:100], p[:100])  # Only plot the first 30 frequencies
+ax.set_xlabel('Frequency (Hz)')
+ax.set_ylabel('Power')
+ax.set_title('Power spectrum of signal 1')
+
+# Compute the coherence between the two timeseries
+f, coh = coherence(signal1, signal2, fs=sfreq, nperseg=100, noverlap=64)
+ax = axes[1][1]
+ax.plot(f[:50], coh[:50])
+ax.set_xlabel('Frequency (Hz)')
+ax.set_ylabel('Coherence')
+ax.set_title('Coherence between the timeseries')
+
+fig.tight_layout()
+
+###############################################################################
+# Now we put the signals at two locations on the cortex. We construct a
+# :class:`mne.SourceEstimate` object to store them in.
+
+# The locations on the cortex where the signal will originate from. These
+# locations are indicated as vertex numbers.
+source_vert1 = 146374
+source_vert2 = 33830
+
+# Construct a SourceEstimate object that describes the signal at the cortical
+# level.
+stc = mne.SourceEstimate(
+    np.vstack((signal1, signal2)),  # The two signals
+    vertices=[[source_vert1], [source_vert2]],  # Their locations
+    tmin=0,
+    tstep=1. / sfreq,
+    subject='sample',  # We use the brain model of the MNE-Sample dataset
+)
+
+###############################################################################
+# Before we simulate the sensor-level data, let's define a signal-to-noise
+# ratio. You are encouraged to play with this parameter and see the effect of
+# noise on our results.
+snr = 1  # Signal-to-noise ratio. Decrease to add more noise.
+
+###############################################################################
+# Now we run the signal through the forward model to obtain simulated sensor
+# data. To save computation time, we'll only simulate gradiometer data. You can
+# try simulating other types of sensors as well.
+
+# Load the forward model
+fwd = mne.read_forward_solution(fwd_fname)
+
+# Use only gradiometers
+fwd = mne.pick_types_forward(fwd, meg='grad', eeg=False, exclude='bads')
+
+# Create an info object that holds information about the sensors (their
+# location, etc.).
+info = mne.create_info(fwd['info']['ch_names'], sfreq, ch_types='grad')
+info.update(fwd['info'])
+
+# To simulate the data, we need a version of the forward solution where each
+# source has a "fixed" orientation, i.e. pointing orthogonally to the surface
+# of the cortex.
+fwd_fixed = mne.convert_forward_solution(fwd, force_fixed=True)
+
+# Now we can run our simulated signal through the forward model, obtaining
+# simulated sensor data.
+sensor_data = mne.apply_forward_raw(fwd_fixed, stc, info).get_data()
+
+# We're going to add some noise to the sensor data
+noise = rand.randn(*sensor_data.shape)
+
+# Scale the noise to be in the ballpark of MEG data
+noise_scaling = np.linalg.norm(sensor_data) / np.linalg.norm(noise)
+noise *= noise_scaling
+
+# Mix noise and signal with the given signal-to-noise ratio.
+sensor_data = snr * sensor_data + noise
```

Comment:
This whole block could be replaced by a suitable call to `mne.simulation.simulate_raw`. And I see that you later need a segment with signal and one with noise, you can e.g. change your STC such that the first half has signal and the second half is all zeros.

Reply:
I tried doing that when I first wrote the tutorial, but ended up not really liking how `simulate_raw` insists of creating its own forward operator, which adds loading BEM meshes and a slew of parameters to the tutorial.

Reply:
I think my suggestion would require 3 filenames (bem, source space, and trans) plus one call to `simulate_raw`. The current code requires one filename (the forward) and eight operations: 1) load the forward, 2) hack the forward info, 3) pick forward channels, 4) convert the forward to surf ori, 5) create the noise, 6) scale the noise to be in the ballpark of reasonable values, 7) call `apply_forward_raw`, 8) add the noise. The former seems simpler than the latter to me.

Either way, a more principled way to do the noise gen would be to load the empty room or baseline cov and use that to generate the noise. You can do it manually with `multivariate_normal` if you wish to stick with these manual computations, otherwise you can let `simulate_raw` do it for you by passing it the cov filename (I think? maybe you need to read the cov...).

In principle this noise cov can also substitute for your "noise" step, so you wouldn't even need to use the first half / last half trick I mentioned above if you do want to try `simulate_raw`. If you want to show how to compute it, though, all you'd really need to do is add a zero -0.2-0 baseline segment in your STC, and use that to compute the baseline cov as your noise cov.